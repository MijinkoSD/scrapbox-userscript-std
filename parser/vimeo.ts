const vimeoRegExp = /https?:\/\/vimeo\.com\/([0-9]+)/i;

/** vimeoのURLからvideo IDを取得する
 *
 * @param url
 * @return video ID vimeoのURLでなければ`undefined`を返す
 */
export const parseVimeo = (url: string): string | undefined => {
  const matches = url.match(vimeoRegExp);
  if (!matches) return undefined;
  return matches[1];
};
