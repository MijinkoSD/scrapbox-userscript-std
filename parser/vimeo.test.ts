import { parseVimeo } from "./vimeo.ts";
import { assertSnapshot } from "../deps/testing.ts";

Deno.test("vimeo links", async (t) => {
  await t.step("is", async (t) => {
    await assertSnapshot(
      t,
      parseVimeo("https://vimeo.com/121284607"),
    );
  });

  await t.step("is not", async (t) => {
    await assertSnapshot(
      t,
      parseVimeo(
        "https://gyazo.com/da78df293f9e83a74b5402411e2f2e01",
      ),
    );
    await assertSnapshot(
      t,
      parseVimeo(
        "ほげほげ",
      ),
    );
    await assertSnapshot(
      t,
      parseVimeo(
        "https://yourtube.com/watch?v=rafere",
      ),
    );
    await assertSnapshot(
      t,
      parseVimeo(
        "https://example.com",
      ),
    );
  });
});
