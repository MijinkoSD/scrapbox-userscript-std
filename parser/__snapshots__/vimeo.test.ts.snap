export const snapshot = {};

snapshot[`vimeo links > is 1`] = `"121284607"`;

snapshot[`vimeo links > is not 1`] = `undefined`;

snapshot[`vimeo links > is not 2`] = `undefined`;

snapshot[`vimeo links > is not 3`] = `undefined`;

snapshot[`vimeo links > is not 4`] = `undefined`;
